modules/sound.nix

```diff
@@ -11,17 +11,34 @@
   };
 
   config = lib.mkIf config.jovian.enableSoundSupport {
-    environment.variables.ALSA_CONFIG_UCM2 = "/run/current-system/sw/share/alsa/ucm2";
-    systemd.user.services.pulseaudio.environment.ALSA_CONFIG_UCM2 = config.environment.variables.ALSA_CONFIG_UCM2;
-    systemd.services.pulseaudio.environment.ALSA_CONFIG_UCM2 = config.environment.variables.ALSA_CONFIG_UCM2;
+    hardware.pulseaudio.enable = false;
```

Comment:
UX-wise, is there a good way we could fail gracefully when `pulseaudio` is enabled, rather than *hope* that it's not `mkForce`'d to true?

Thinking maybe using `config.assertions`. Something like:

```nix
{
  assertions = [
    { assertion = !config.hardware.pulseaudio.enable; message = "Sound support for the steam deck requires pipewire. Remove usage of pulseaudio from your configuration."; }
  ];
}
```

Maybe mixed with a `mkDefault false` or `mkOverride 51 false`? The former would probably do nothing, so eh, not useful. The latter would be just less forced than `mkForce`. This means that users that "just" enable pulseaudio wouldn't see the assertion, but users that `mkForce` pulseaudio` would see it. Setting it at `mkForce` would cause a conflict in the option values and no assertion would explain the situation to the users.

Or am I missing a mechanism in the NixOS modules system that would be better?